test/python/quantum_info/operators/test_scalar_op.py

```diff
@@ -0,0 +1,636 @@
+                    self.assertEqual(val.output_dims(), dims)
+                    self.assertAlmostEqual(val.coeff, target)
+
+                with self.subTest(msg='{} @ {}([0])'.format(op1, op2)):
+                    val = op1 @ op2([0])
+                    if coeff1 is None:
+                        target = coeff2
+                    elif coeff2 is None:
+                        target = coeff1
+                    else:
+                        target = coeff1 * coeff2
+                    self.assertTrue(isinstance(val, ScalarOp))
+                    self.assertEqual(val.input_dims(), dims)
+                    self.assertEqual(val.output_dims(), dims)
+                    self.assertAlmostEqual(val.coeff, target)
+
+                with self.subTest(msg='{} @ {}([1])'.format(op1, op3)):
+                    val = op1 @ op3([1])
+                    if coeff1 is None:
+                        target = coeff2
+                    elif coeff2 is None:
+                        target = coeff1
+                    else:
+                        target = coeff1 * coeff2
+                    self.assertTrue(isinstance(val, ScalarOp))
+                    self.assertEqual(val.input_dims(), dims)
+                    self.assertEqual(val.output_dims(), dims)
+                    self.assertAlmostEqual(val.coeff, target)
+
+                with self.subTest(msg='{} * {}([0])'.format(op1, op2)):
+                    val = op1 * op2([0])
+                    if coeff1 is None:
+                        target = coeff2
+                    elif coeff2 is None:
+                        target = coeff1
+                    else:
+                        target = coeff1 * coeff2
+                    self.assertTrue(isinstance(val, ScalarOp))
+                    self.assertEqual(val.input_dims(), dims)
+                    self.assertEqual(val.output_dims(), dims)
+                    self.assertAlmostEqual(val.coeff, target)
+
+                with self.subTest(msg='{} * {}([1])'.format(op1, op3)):
+                    val = op1 * op3([1])
+                    if coeff1 is None:
+                        target = coeff2
+                    elif coeff2 is None:
+                        target = coeff1
+                    else:
+                        target = coeff1 * coeff2
+                    self.assertTrue(isinstance(val, ScalarOp))
+                    self.assertEqual(val.input_dims(), dims)
+                    self.assertEqual(val.output_dims(), dims)
+                    self.assertAlmostEqual(val.coeff, target)
+
+    def test_compose_operator(self):
+        """Test compose and dot methods with ScalarOp and Operator."""
+        for coeff in [None, 1, -3.1, 1 + 3j]:
+            for label in ['II', 'XX', 'YY', 'ZZ']:
+                dim = 4
+                iden = ScalarOp(dim, coeff=coeff)
+                op = Operator.from_label(label)
+
+                with self.subTest(msg='{}.compose(Operator({}))'.format(iden, label)):
```

Comment:
@1ucian0 can something like this use the `combine` decorator of tests, instead of using subTest? or is subTest fine?

Reply:
I would say both, at least in some scenarios.  I had trouble sleeping so I refactor the tests a bit in 73d5cc5eb. I think they are much easier to read now.

Reply:
I like using subTest because it shows each failing subtest as a separate test failure when something goes wrong.

Reply:
`combine` also allows that and, in addition, you can run the failing test independently of the rest of the cases. So your breakpoints for debugging are easier set.